test/verify/testsuite-prepare

```diff
@@ -1,87 +1,66 @@
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 
-silent()
-{
-       "$@" > /dev/null 2> /dev/null
-       return $?
-}
+import argparse
+import os
+import shutil
+import sys
+import imp
 
-usage()
-{
-    echo "usage: testsuite-prepare [--clean] [--quick] [--verbose]"
-}
+basepath = os.path.join(os.path.dirname(os.path.abspath(__file__)),"..")
+if not basepath in sys.path:
+    sys.path.insert(1, basepath)
+from common import testinfra
+vminstall = imp.load_source('vminstall', os.path.join(basepath,"vm-install"))
+vmdownload = imp.load_source('vmdownload', os.path.join(basepath,"vm-download"))
 
+image_list="ipa openshift fedora-stock"
 
-args=$(getopt -o "h,v,q" -l "help,verbose,quick" -- "$@")
-eval set -- "$args"
-while [ $# -gt 0 ]; do
-    case $1 in
-    -q|--quick)
-        vm_install_opts="$vm_install_opts --quick"
-        ;;
-    -v|--verbose)
-        vm_install_opts="$vm_install_opts --verbose"
-        ;;
-    -h|--help)
-        usage
-        exit 0
-        ;;
-    --)
-        shift
-        break
-        ;;
-    esac
-    shift
-done
+parser = argparse.ArgumentParser(
+        description='Prepare testig environment, download images and build and cinstall cockpit',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+parser.add_argument('-v', '--verbose', action='store_true', help='Display verbose progress details')
+parser.add_argument('-n', '--noinstall', action='store_true', help='Do not install cockpit to base image')
+parser.add_argument('-f', '--force', action='store_true', help='Force update of images')
+parser.add_argument('image', nargs='?', default=testinfra.DEFAULT_IMAGE, help='The image to use')
+args = parser.parse_args()
 
-# The Atomic variants can't build their own packages, so we build in
-# their non-Atomic siblings.  For example, fedora-atomic is build
-# in fedora-23
-#
-if [ "$TEST_OS" = "fedora-atomic" ]; then
-    BUILD_OS="fedora-23"
-elif [ "$TEST_OS" = "rhel-atomic" ]; then
-    BUILD_OS="rhel-7"
-else
-    BUILD_OS="$TEST_OS"
-fi
+test_os = args.image
+build_os = test_os
 
-$BASE/vm-reset
-$BASE/vm-download \
-    $TEST_OS \
-    $BUILD_OS \
-    ipa \
-    openshift \
-    fedora-stock
+if test_os == "fedora-atomic":
+    build_os = "fedora-23"
+elif test_os == "rhel-atomic":
+    build_os = "rhel-7"
 
-install_failed=no
-if ! $BASE/vm-install $vm_install_opts --build-image $BUILD_OS; then
-    install_failed=yes
-fi
+# vm-reset replacement
+shutil.rmtree("%s/tmp/run" % basepath)
+os.makedirs("%s/tmp/run" % basepath)
```

Comment:
I tried to avoid to call some os.system calls, in that case that in vm-reset are only 2 commands I've decided to replace it directly